application/extensions/SideBarWidget/SideBarWidget.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+class SideBarWidget extends CWidget
+{
+    /** Initializes the widget */
+    public function init(): void
+    {
+        $this->registerClientScript();
+    }
+
+    /** Executes the widget
+     * @throws CException
+     */
+    public function run(): void
+    {
+        $this->renderActions();
+    }
+
+    /**
+     * Renders the sidebar menu on the dashboard page
+     * @throws CException
+     */
+    public function renderActions(): void
+    {
+        $boxes = Box::model()->findAll(['order' => 'position ASC']);
+        $boxesData = [];
+        foreach ($boxes as $box) {
+            $boxData['position'] = $box->position;
+            if (!preg_match("/^(http|https)/", $box->url)) {
+                $boxData['url'] = App()->createUrl($box->url);
+                $boxData['external'] = false;
+            } else {
+                $boxData['url'] = $box->url;
+                $boxData['external'] = true;
+            }
+            $boxData['title'] = $box->title;
+            $boxData['ico'] = $box->getIconName();
+            $boxData['description'] = $box->desc;
+
+            // default permission if usergroup is not within expected values
+            $canSeeBox = false;
+            if (Permission::model()->hasGlobalPermission('superadmin', 'read')) {
+                $canSeeBox = true;
+            }
+            // everyone can see the box
+            if ($box->usergroup === -1) {
```

Comment:
checking usergroup value  for int, but value is stored as string in db.

This made my restricted user having no sidebar at all.

Reply:
The usergroup is of type int(11), not string, so it shouldn't be a problem, right?

Reply:
$box->usergroup is string.
In my case, I test with a permission restricted user, but I have boxes with usergroup: '-1'
Those should be shown to the user, but are not because the condition ist strict on type int

Reply:
Works fine now.
It's a yii thing, although the db col is integer add the model also declares the property as integer, 
it come as a string in the ActiveRecord obj.